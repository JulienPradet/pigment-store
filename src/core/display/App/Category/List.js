import React from 'react'
import {Link} from 'react-router'
import * as LinkList from '../util/View/LinkList'

export default ({category, pathname}) => {
  return <LinkList.Container>
    <LinkList.Title>Subcategories</LinkList.Title>
    <LinkList.Content>
      <ul>
        {category.categories.map(({name}) => <li key={name}>
          <Link to={`${pathname}/category-${name}`}>{name}</Link>
        </li>)}
      </ul>
    </LinkList.Content>
  </LinkList.Container>
}
