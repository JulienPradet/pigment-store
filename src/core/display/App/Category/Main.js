import React from 'react'
import {Match} from 'react-router'
import CurrentCategory from './Current'
import Component from '../Component/Main'

const Category = ({category, pathname}) => {
  return <div>
    <Match
      exactly
      pattern={`${pathname}`}
      render={(matchProps) => <CurrentCategory category={category} pathname={pathname} />}
    />
    <Match
      pattern={`${pathname}/category-:categoryName`}
      render={({pathname, params}) => {
        const categoryName = params.categoryName
        const subCategory = category.categories.find((category) => category.name === categoryName)
        return <Category category={subCategory} pathname={pathname} />
      }}
    />
    <Match
      pattern={`${pathname}/component-:componentName`}
      render={({pathname, params}) => {
        const componentName = params.componentName
        const component = category.components.find((component) => component.name === componentName)
        return <Component component={component} pathname={pathname} />
      }}
    />
  </div>
}

export default Category
