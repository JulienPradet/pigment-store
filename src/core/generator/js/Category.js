import Component from './Component'

export default class Category {
  constructor (name) {
    this.name = name
    this.subCategories = []
    this.components = []
  }

  addCategory ({name, category}) {
    if (!(category instanceof Category)) {
      console.warn(`The category should be an instance of Category. ${typeof category} given.`)
    } else {
      this.subCategories.push({
        name,
        category
      })
    }
    return this
  }

  addComponent ({name, component}) {
    if (!(component instanceof Component)) {
      console.warn(`The component should be an instance of Component. ${typeof component} given.`)
    } else {
      this.components.push({
        name,
        component
      })
    }
    return this
  }

  render () {
    return `
      {
        name: '${this.name}',
        categories: [
          ${this.subCategories
            .map(({name, category}) => `{
              name: '${name}',
              category: ${category.render()}
            }`)
            .join(',')}
        ],
        components: [
          ${this.components
            .map(({name, component}) => `${component.render()}`)
            .join(',')}
        ]
      }
    `
  }
}
