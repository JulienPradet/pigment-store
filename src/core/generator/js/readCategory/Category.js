const Component = require('./Component')

const renderString = (string) => {
  return string && `'${string.replace(/\\/g, '\\\\').replace(/'/g, '\\\'').replace(/\n/g, '\\n')}'`
}

module.exports = class Category {
  constructor (name) {
    this.name = name
    this.subCategories = []
    this.components = []
    this.description = null
  }

  addCategory ({name, category}) {
    if (!(category instanceof Category)) {
      console.warn(`The category should be an instance of Category. ${typeof category} given.`)
    } else {
      this.subCategories.push({
        name,
        category
      })
    }
    return this
  }

  addComponent ({name, component}) {
    if (!(component instanceof Component)) {
      console.warn(`The component should be an instance of Component. ${typeof component} given.`)
    } else {
      this.components.push({
        name,
        component
      })
    }
    return this
  }

  setDescription (description) {
    this.description = description
    return this
  }

  render () {
    const render = `{
      name: ${renderString(this.name)},
      description: ${renderString(this.description)},
      categories: [${this.subCategories.map(
        ({name, category}) => category.render()
      ).join(',')}],
      components: [${this.components.map(
        ({name, component}) => component.render()
      ).join(',')}]
    }`

    return render
  }
}
