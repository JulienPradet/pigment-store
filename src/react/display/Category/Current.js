import React from 'react'
import {Link} from 'react-router'
import {PageTitle} from '../util/View/Title'
import CategoryList from './List'
import ComponentList from '../Component/List'
import Card from '../util/View/Card'
import Markdown from '../util/View/Markdown'

class CurrentCategory extends React.Component {
  constructor (props) {
    super()
    document.title = props.category.name
  }

  render () {
    const {category, pathname} = this.props

    const title = <PageTitle>
      <Link to={`${pathname}`}>{category.name}</Link>
    </PageTitle>

    return <div>
      <Card title={title}>
        {category.description ? <Markdown>{category.description}</Markdown> : null}
        {category.categories.length > 0 ? <CategoryList category={category} pathname={pathname} /> : null}
      </Card>

      <ComponentList category={category} pathname={pathname} />
    </div>
  }
}

export default CurrentCategory
