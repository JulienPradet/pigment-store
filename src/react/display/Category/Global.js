import React from 'react'

import {PageTitle} from '../util/View/Title'
import CategoryList from './List'
import ComponentList from '../Component/List'
import Card from '../util/View/Card'
import Markdown from '../util/View/Markdown'

const Category = ({category, prefix}) => {
  const title = <PageTitle>
    {category.name}
  </PageTitle>

  return <div>
    <Card title={title}>
      {category.description ? <Markdown>{category.description}</Markdown> : null}
      {category.categories.length > 0 ? <CategoryList category={category} prefix={prefix} /> : null}
    </Card>

    <ComponentList category={category} prefix={prefix} />
  </div>
}

export default ({category, prefix}) => {
  const result = class extends React.Component {
    constructor () {
      super()
      document.title = category.name
    }

    render () {
      return <Category {...this.props} category={category} prefix={prefix} />
    }
  }

  result.displayName = 'Category'

  return result
}
