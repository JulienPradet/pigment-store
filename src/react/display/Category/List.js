import React from 'react'
import {Link} from 'react-router'

import {makePath} from '../router'
import * as LinkList from '../util/View/LinkList'

export default ({category, prefix}) => {
  return <LinkList.Container>
    <LinkList.Title>Subcategories</LinkList.Title>
    <LinkList.Content>
      <ul>
        {category.categories.map(({name}) => <li key={name}>
          <Link to={makePath(prefix, category.name, name)}>{name}</Link>
        </li>)}
      </ul>
    </LinkList.Content>
  </LinkList.Container>
}
