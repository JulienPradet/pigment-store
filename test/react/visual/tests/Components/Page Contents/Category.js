import React from 'react'
import PigmentStore from 'pigment-store'
import Category from '../../../../../../src/core/display/App/Category/Current'
import rendererDecorator from '../../__decorators__/.renderer.decorator'
import routerDecorator from '../../__decorators__/.router.decorator'
import configDecorator from '../../__decorators__/.config.decorator'
import displayOptionsDecorator from '../../__decorators__/.displayOptions.decorator'

import fixtures from '../__fixtures__/index'

export default PigmentStore.React.describe('Category', Category)
  .addDecorator(rendererDecorator)
  .addDecorator(routerDecorator)
  .addDecorator(configDecorator)
  .addDecorator(displayOptionsDecorator)
  .featureJsx('Default', <Category category={fixtures} pathname='' />)
  .featureJsx(
    'With components',
    (
      <Category category={fixtures.categories[0]} />
    )
  )
