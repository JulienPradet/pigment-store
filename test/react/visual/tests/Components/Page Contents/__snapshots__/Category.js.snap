exports[`Category Default 1`] = `
<div>
  <div
    className="container"
    id={undefined}>
    <div
      className="titleContainer"
      onClick={undefined}>
      <div
        className="title">
        <div
          className="titleContent">
          <h1>
            <a
              className=""
              href=""
              onClick={[Function]}
              style={Object {}}>
              Index Category
            </a>
          </h1>
        </div>
      </div>
    </div>
    <div>
      <div
        className="collapsibleContent">
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>The main description of the styleguide.</p>
            ",
            }
          } />
        <div
          className="container">
          <div
            className="title">
            Subcategories
          </div>
          <div
            className="content">
            <ul>
              <li>
                <a
                  className=""
                  href="/category-Atom"
                  onClick={[Function]}
                  style={Object {}}>
                  Atom
                </a>
              </li>
              <li>
                <a
                  className=""
                  href="/category-Molecules"
                  onClick={[Function]}
                  style={Object {}}>
                  Molecules
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;

exports[`Category With components 1`] = `
<div>
  <div
    className="container"
    id={undefined}>
    <div
      className="titleContainer"
      onClick={undefined}>
      <div
        className="title">
        <div
          className="titleContent">
          <h1>
            <a
              className=""
              href="undefined"
              onClick={[Function]}
              style={Object {}}>
              Atom
            </a>
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="container"
      id={undefined}>
      <div
        className="titleContainer"
        onClick={[Function]}>
        <div
          className="title">
          <div
            className="titleContent">
            <h2>
              <a
                className=""
                href="undefined/component-Title"
                onClick={[Function]}
                style={Object {}}>
                Title
              </a>
            </h2>
          </div>
          <div
            className="titleButton">
            <button
              onClick={[Function]}>
              <i
                className="material-icons">
                keyboard_arrow_down
              </i>
            </button>
          </div>
        </div>
        <div
          className="content">
          <div
            style={
              Object {
                "height": "100%",
                "margin": "0 auto",
                "width": "100%",
              }
            }>
            <iframe
              src="iframe.html#undefined/component-Title/feature-Default"
              style={
                Object {
                  "border": 0,
                  "height": "100px",
                  "transform": "scale(1)",
                  "transformOrigin": "top left",
                  "width": "100%",
                }
              } />
          </div>
        </div>
      </div>
      <div />
    </div>
    <div
      className="container"
      id={undefined}>
      <div
        className="titleContainer"
        onClick={[Function]}>
        <div
          className="title">
          <div
            className="titleContent">
            <h2>
              <a
                className=""
                href="undefined/component-Avatar"
                onClick={[Function]}
                style={Object {}}>
                Avatar
              </a>
            </h2>
          </div>
          <div
            className="titleButton">
            <button
              onClick={[Function]}>
              <i
                className="material-icons">
                keyboard_arrow_down
              </i>
            </button>
          </div>
        </div>
        <div
          className="content">
          <div
            style={
              Object {
                "height": "100%",
                "margin": "0 auto",
                "width": "100%",
              }
            }>
            <iframe
              src="iframe.html#undefined/component-Avatar/feature-Default"
              style={
                Object {
                  "border": 0,
                  "height": "100px",
                  "transform": "scale(1)",
                  "transformOrigin": "top left",
                  "width": "100%",
                }
              } />
          </div>
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
